src/LinuxTracing/ThreadStateManager.cpp

```diff
@@ -40,24 +43,34 @@ void ThreadStateManager::OnNewTask(uint64_t timestamp_ns, pid_t tid) {
     ORBIT_ERROR("Processed task:task_newtask but thread %d was already known", tid);
     return;
   }
-  tid_open_states_.insert_or_assign(tid, OpenState{kNewState, timestamp_ns});
+  tid_open_states_.insert_or_assign(
+      tid, OpenState{kNewState, timestamp_ns, was_created_by_tid, was_created_by_pid,
+                     orbit_grpc_protos::ThreadStateSlice::WakeupReason::
+                         ThreadStateSlice_WakeupReason_kCreated});
 }
 
-std::optional<ThreadStateSlice> ThreadStateManager::OnSchedWakeup(uint64_t timestamp_ns,
-                                                                  pid_t tid) {
+std::optional<ThreadStateSlice> ThreadStateManager::OnSchedWakeup(uint64_t timestamp_ns, pid_t tid,
+                                                                  pid_t was_blocked_by_tid,
+                                                                  pid_t was_blocked_by_pid) {
   static constexpr ThreadStateSlice::ThreadState kNewState = ThreadStateSlice::kRunnable;
 
   auto open_state_it = tid_open_states_.find(tid);
   if (open_state_it == tid_open_states_.end()) {
     ORBIT_ERROR("Processed sched:sched_wakeup but previous state of thread %d is unknown", tid);
-    tid_open_states_.insert_or_assign(tid, OpenState{kNewState, timestamp_ns});
+    tid_open_states_.insert_or_assign(
+        tid, OpenState{kNewState, timestamp_ns, was_blocked_by_tid, was_blocked_by_pid,
+                       orbit_grpc_protos::ThreadStateSlice::WakeupReason::
+                           ThreadStateSlice_WakeupReason_kUnblocked});
```

Comment:
`orbit_grpc_protos::ThreadStateSlice::WakeupReason::kUnblocked` or `orbit_grpc_protos::ThreadStateSlice::kUnblocked` (not sure which one works)